packages/wdio-allure-reporter/src/utils.js

```diff
@@ -0,0 +1,46 @@
+import process from 'process'
+import {testStatuses} from './constants'
+/**
+ * Get allure test status by TestStat object
+ * @param test {Object} - TestStat object
+ * @param config {Object} - wdio config object
+ * @private
+ */
+export const getTestStatus = (test, config) => {
+    if (config.framework === 'jasmine') {
+        return testStatuses.FAILED
+    }
+
+    if (test.error.name) {
+        return test.error.name === 'AssertionError' ? testStatuses.FAILED : testStatuses.BROKEN
+    }
+
+    const stackTrace = test.error.stack.trim()
+    return stackTrace.startsWith('AssertionError') ? testStatuses.FAILED : testStatuses.BROKEN
+
+}
+
+/**
+ * Check is object is empty
+ * @param object {Object}
+ * @private
+ */
+export const isEmpty = (object) => !object || Object.keys(object).length === 0
+
+/**
+ * Filter unnecessary mocha hooks
+ * @param title {String} - hook title
+ * @returns {boolean}
+ * @private
+ */
+export const ignoredHooks = title => ['"before all" hook', '"after all" hook', '"before each" hook', '"after each" hook'].some(hook => title.includes(hook))
```

Comment:
Maybe worth moving the list into a const (maybe even in a separate constants.js file like in the other packages). Also do the ignored hooks similar in Mocha/Jasmine and Cucumber?

Reply:
yes, I've moved them.

>Also do the ignored hooks similar in Mocha/Jasmine and Cucumber?
Actually it is only for Mocha. I don't why but mocha runner spams hook events (with Before\AfterEach Before\AfterAll in title) even spec file don't have any mocha hooks...

I don't test it with Cucumber because there is no v5 wdio-cucumber. When it will be done I will port additional features for reporter and make some tests.